test/unit-test/ice_api_private/ice_api_private_utest.c

```diff
@@ -278,3 +280,222 @@ void test_iceFinalizeStunPacket_StunError_SmallBuffer( void )
+
+    TEST_ASSERT_EQUAL( 0,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Same Transport Address check fail functionality.
+ */
+void test_iceIsSameTransportAddress_Diff_Port( void )
+{
+    IceTransportAddress_t transportAddress1;
+    IceTransportAddress_t transportAddress2;
+    uint8_t ipAddress[] = { 0xC0, 0xA8, 0x01, 0x64 };        /* "192.168.1.100". */
+    uint8_t result;
+
+    transportAddress1.port = 3478;
+    transportAddress1.family = 1;
+    memcpy( ( void * ) &( transportAddress1.address[ 0 ] ),
+            ( const void * ) ipAddress,
+            sizeof( ipAddress ) );
+
+    transportAddress2.port = 2002;
+    transportAddress2.family = 1;
+    memcpy( ( void * ) &( transportAddress2.address[ 0 ] ),
+            ( const void * ) ipAddress,
+            sizeof( ipAddress ) );
+
+
+    result = Ice_IsSameTransportAddress( &( transportAddress1 ),
+                                         &( transportAddress2 ) );
+
+    TEST_ASSERT_EQUAL( 0,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Same Transport Address check fail functionality.
+ */
+void test_iceIsSameTransportAddress_Diff_Address( void )
+{
+    IceTransportAddress_t transportAddress1;
+    IceTransportAddress_t transportAddress2;
+    uint8_t ipAddress[] = { 0xC0, 0xA8, 0x01, 0x64 };        /* "192.168.1.100". */
+    uint8_t ipAddress2[] = { 0x78, 0xA8, 0x01, 0x6E };       /* "192.168.1.110". */
+    uint8_t result;
+
+    transportAddress1.port = 3478;
+    transportAddress1.family = 1;
+    memcpy( ( void * ) &( transportAddress1.address[ 0 ] ),
+            ( const void * ) ipAddress,
+            sizeof( ipAddress ) );
+
+    transportAddress2.port = 3478;
+    transportAddress2.family = 1;
+    memcpy( ( void * ) &( transportAddress2.address[ 0 ] ),
+            ( const void * ) ipAddress2,
+            sizeof( ipAddress2 ) );
+
+
+    result = Ice_IsSameTransportAddress( &( transportAddress1 ),
+                                         &( transportAddress2 ) );
+
+    TEST_ASSERT_EQUAL( 0,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Compute Candidate Priority check fail functionality.
+ */
+void test_iceComputeCandidatePriority_Invalid( void )
+{
+    IceCandidateType_t candidateType;
+    uint8_t isPointToPoint;
+    uint64_t result;
+
+    candidateType = 5; /* Unkown Type */
+    isPointToPoint = 1;
+
+    /* priority = (2^24)*(type preference) +
+     *   (2^8)*(local preference) +
+     *   (2^0)*(256 - component ID) */
+
+
+    /* ( 2^24 ) * ( 0 ) + ( 2^8 ) * ( 0 ) +  255     =   255  */
+
+    result = Ice_ComputeCandidatePriority( candidateType,
+                                           isPointToPoint );
+
+    TEST_ASSERT_EQUAL( 255,
```

Comment:
Why does it return 255 when it's an unknown type? Is 255 a magic code for error in this function?

Reply:
No, that I clarified with Gaurav too, in the webRTC SDK as well they are keeping the type preference = 0 for both a unkown type and turn type , I thought if we could keep it -1 or some other value to distinguish would be better. But since unkown type can never come. Gaurav said it doesn't matter much.